doc-src/ch-test_desc.rst

```diff
@@ -3,159 +3,178 @@ Synopsis
-    Defaults: :code:`$CH_TEST_IMGDIR` if set, or :code:`/var/tmp/img`.
+  a specific :code:`.bats` file
+    Run the tests in that file. **(Not yet implemented.)**
 
-  :code:`--perm-dir DIR`
-    Specify directory :code:`DIR` for file permission enforcement testing.
-    Requires :code:`--sudo`. Default: :code:`CH_TEST_PERMDIRS` if set, or
-    :code:`skip`.
+If :code:`PHASE` is not specified, run :code:`build`, :code:`run`, and
+:code:`examples` phases in that order.
 
-    :code:`--perm-dir` can be set multiple times to create testing fixtures
-    in multiple places.
+Scope is specified with:
 
-  :code:`-s`, :code:`--scope [quick|standard|full]`
-    Run tests with given scope (speed vs. thoroughness). Default:
+  :code:`-s`, :code:`--scope SCOPE`
+    :code:`SCOPE` must be one of the following; the default is
     :code:`standard`.
 
-    * :code:`quick` - Test the most important subset of Charliecloud
-      functionality. Handy for development. Estimated completion time (clean
-      image cache): 60-120 seconds.
+    * :code:`quick`: Most important subset of workflow. Handy for development.
+      Completion time: 1–2 minutes.
 
-    * :code:`standard` - Test all Charliecloud functionality and a selection of
-      more important examples. Estimated completion time (clean image cache):
-      5-10 minutes.
+    * :code:`standard`: All tested workflow functionality and a selection of
+      more important examples. Completion time: 5–10 minutes.
 
-    * :code:`full` - Run all available tests and examples. Estimated
-      completion time (clean image cache): 1-2 hours.
+    * :code:`full`: All available tests, including all examples. Completion
+      time, hot cache: 7–15 minutes; cold cache: 1–2 hours.
 
-  :code:`--sudo`
-    Run an extra set of tests, e.g., file permission enforcement tests,
-    privilege escalation tests, etc. Requires :code:`sudo` capabilities.
+Additional arguments:
 
-  :code:`--tar-dir DIR`
-    Store image tarballs from :code:`build` phase in directory :code:`DIR`.
-    Defaults: :code:`$CH_TEST_TARDIR` if set, or :code:`/var/tmp/tar`.
+  :code:`-b`, :code:`--builder BUILDER`
+    Image builder to use. See ch-build(1) for how the default is selected.
 
-Storage
-=======
+  :code:`--dry-run`
+    Print summary of what would be tested and then exit.
 
-The test suite requires a few tens of GB of storage for test fixtures:
+  :code:`-h`, :code:`--help`
+    Print usage and then exit.
 
-* Builder storage (e.g., layer cache). This goes wherever the builder puts it.
+  :code:`--img-dir DIR`
+    Set unpacked images directory to :code:`DIR`. In a multi-node allocation,
+    this directory may not be shared between nodes. Default:
+    :code:`$CH_TEST_IMGDIR` if set, or :code:`/var/tmp/img`.
 
-* Compressed image directory, i.e., :code:`--tar-dir DIR`.
+  :code:`--pack-dir DIR`
+    Set packed images directory to :code:`DIR`. Default:
+    :code:`$CH_TEST_TARDIR` if set, or :code:`/var/tmp/pack`.
```

Comment:
Perhaps we should go ahead and update `CH_TEST_TARDIR` to `CH_TEST_PACKDIR` as well? I know it is long, but I'm also wondering if `CH_TEST_IMGDIR` should be `CH_TEST_UNPACKDIR`.

Reply:
I guess the reasoning to leave the variables along is that it's a pretty pervasive change that alters an existing API, so I wanted to leave it alone for now, in case the new vocabulary is not satisfactory.